dotnet/targets/Xamarin.Shared.Sdk.props

```diff
@@ -173,6 +164,19 @@
 		<!-- For None link mode we also need to set TrimMode for all assemblies. This is done later -->
 	</PropertyGroup>
 
+	<PropertyGroup>
+		<!--
+			With NativeAOT we want to suppress trim warnings coming from ILLink and enable them only for ILC.
+			For this reason, in case of NativeAOT, we set SuppressTrimAnalysisWarnings to true by default and store the overwriten default in
+			_OriginalSuppressTrimAnalysisWarnings property, which is later used to properly configure warning suppression for ILC.
+		-->
+		<_OriginalSuppressTrimAnalysisWarnings>$(SuppressTrimAnalysisWarnings)</_OriginalSuppressTrimAnalysisWarnings>
+		<SuppressTrimAnalysisWarnings Condition="'$(_UseNativeAot)' == 'true'">true</SuppressTrimAnalysisWarnings>
+		<!-- Otherwise suppress trimmer warnings unless we're trimming all assemblies -->
+		<SuppressTrimAnalysisWarnings Condition="'$(SuppressTrimAnalysisWarnings)' == '' And '$(TrimMode)' == 'full'">false</SuppressTrimAnalysisWarnings>
```

Comment:
This is basically what is in the Android workload right now, but I noticed you can also use `TrimMode=link`:

![image](https://github.com/xamarin/xamarin-macios/assets/840039/9e4e9fa9-5243-46f6-bede-5fa9be2dd6ca)

I wonder if this `TrimMode=link` setting is actually wrong? I can't find this in the docs:

* https://learn.microsoft.com/en-us/dotnet/core/deploying/trimming/trimming-options?pivots=dotnet-8-0#trimming-granularity

And not finding anything in .NET like: `<TrimMode Condition="'$(TrimMode)'=='link'">full</TrimMode>`

Reply:
It shows up if you select `.NET 6.0` at the top of the page: https://learn.microsoft.com/en-us/dotnet/core/deploying/trimming/trimming-options?pivots=dotnet-6-0#trimming-granularity